aws_lambda_powertools/utilities/parser/models/s3.py

```diff
@@ -11,88 +11,319 @@
 
 class S3EventRecordGlacierRestoreEventData(BaseModel):
     lifecycleRestorationExpiryTime: datetime
-    lifecycleRestoreStorageClass: str
+    lifecycleRestoreStorageClass: str = Field(
+        description="Source storage class for restore.",
+        examples=[
+            "standard",
+            "standard_ia",
+            "glacier",
+        ],
+    )
 
 
 class S3EventRecordGlacierEventData(BaseModel):
     restoreEventData: S3EventRecordGlacierRestoreEventData
 
 
 class S3Identity(BaseModel):
-    principalId: str
+    principalId: str = Field(
+        description="Amazon customer ID of the user who caused the event.",
+        examples=[
+            "AIDAJDPLRKLG7UEXAMPLE",
+            "A1YQ72UWCM96UF",
+            "AWS:AIDAINPONIXQXHT3IKHL2",
+        ],
+    )
 
 
 class S3RequestParameters(BaseModel):
     sourceIPAddress: Union[IPvAnyNetwork, Literal["s3.amazonaws.com"]]
 
 
 class S3ResponseElements(BaseModel):
-    x_amz_request_id: str = Field(..., alias="x-amz-request-id")
-    x_amz_id_2: str = Field(..., alias="x-amz-id-2")
+    x_amz_request_id: str = Field(
+        ...,
+        alias="x-amz-request-id",
+        description="Amazon S3 generated request ID.",
+        examples=[
+            "C3D13FE58DE4C810",
+            "D82B88E5F771F645",
+        ],
+    )
+    x_amz_id_2: str = Field(
+        ...,
+        alias="x-amz-id-2",
+        description="ID of the Amazon S3 host that processed the request.",
+        examples=[
+            "FMyUVURIY8/IgAtTv8xRjskZQpcIZ9KG4V5Wp6S7S/JRWeUWerMUE5JgHvANOjpD",
+            "vlR7PnpV2Ce81l0PRw6jlUpck7Jo5ZsQjryTjKlc5aLWGVHPZLj5NeC6qMa0emYBDXOo6QBU0Wo=",
+        ],
+    )
 
 
 class S3OwnerIdentify(BaseModel):
-    principalId: str
+    principalId: str = Field(
+        description="Amazon customer ID of the bucket owner.",
+        examples=[
+            "A3I5XTEXAMAI3E",
+            "A1YQ72UWCM96UF",
+            "AWS:AIDAINPONIXQXHT3IKHL2",
+        ],
+    )
 
 
 class S3Bucket(BaseModel):
-    name: str
+    name: str = Field(
+        description="Name of the Amazon S3 bucket.",
+        examples=[
+            "lambda-artifacts-deafc19498e3f2df",
+            "example-bucket",
+            "sourcebucket",
+        ],
+    )
     ownerIdentity: S3OwnerIdentify
```

Comment:
Can you add a description here? Please let me know if you need help writing them.